python/src/main/python/drivers/runspv.py

```diff
@@ -0,0 +1,944 @@
+              + ANDROID_DEVICE_GRAPHICSFUZZ_DIR + '/' + os.path.basename(amberscript_file) + '"'
+
+        adb_check('logcat -c')
+
+        status = 'UNEXPECTED_ERROR'
+
+        try:
+            result = adb_can_fail(cmd)
+        except subprocess.TimeoutExpired:
+            result = None
+            status = 'TIMEOUT'
+
+        if status != 'TIMEOUT':
+            if result.returncode != 0:
+                status = 'CRASH'
+            else:
+                if adb_can_fail('shell test -f' + device_ssbo).returncode == 0:
+                    status = 'SUCCESS'
+                    adb_check('pull ' + device_ssbo + ' ' + ssbo_output)
+
+        # Grab log:
+        with open(logfile, 'w', encoding='utf-8', errors='ignore') as f:
+            adb_check('logcat -d', stdout=f)
+    else:
+        cmd = 'amber -b ' + ssbo_output + ' -B ' + ssbo_binding + ' ' + amberscript_file + ' > ' + logfile
+        status = 'SUCCESS'
+        try:
+            subprocess.run(cmd, shell=True, timeout=TIMEOUT_RUN).check_returncode()
+        except subprocess.TimeoutExpired:
+            status = 'TIMEOUT'
+        except subprocess.CalledProcessError:
+            status = 'CRASH'
+
+        if status == 'SUCCESS':
+            assert (os.path.isfile(ssbo_output))
+
+        with open(logfile, 'a') as f:
+            f.write('\nSTATUS ' + status + '\n')
+
+    if os.path.isfile(ssbo_output):
+        ssbo_text_to_json(ssbo_output, ssbo_json, json)
+
+    with open(logfile, 'a') as f:
+        f.write('\nSTATUS ' + status + '\n')
+
+    with open(statusfile, 'w') as f:
+        f.write(status)
+
+
+################################################################################
+# Main
+
+
+def some_shader_format_exists(prefix: str, kind: str) -> bool:
+    return os.path.isfile(prefix + '.' + kind) \
+           or os.path.isfile(prefix + '.' + kind + '.asm') \
+           or os.path.isfile(prefix + '.' + kind + '.spv')
+
+
+def multiple_shader_formats_exist(prefix: str, kind: str) -> bool:
+    count = 0
+    if os.path.isfile(prefix + '.' + kind):
+        count += 1
+    if os.path.isfile(prefix + '.' + kind + '.asm'):
+        count += 1
+    if os.path.isfile(prefix + '.' + kind + '.spv'):
+        count += 1
+    return count > 1
+
+
+def pick_shader_format(prefix: str, kind: str) -> str:
+    if multiple_shader_formats_exist(prefix, kind):
+        raise ValueError('More than one of .' + kind + ', .' + kind + '.asm and .' + kind + '.spv are present')
+    if os.path.isfile(prefix + '.' + kind):
+        return prefix + '.' + kind
+    if os.path.isfile(prefix + '.' + kind + '.asm'):
+        return prefix + '.' + kind + '.asm'
+    assert os.path.isfile(prefix + '.' + kind + '.spv')
+    return prefix + '.' + kind + '.spv'
+
+
+def main_helper(args):
+    # TODO: update description to reflect final output.
+    description = 'Run SPIR-V shaders.  Output: ' + LOGFILE_NAME + ', image.png'
```

Comment:
Should we document compute shader output?

Reply:
I made the output png and updated the description here to include image_0.png and ssbo.json.